jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,735 @@
+        }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s@%x{%s}", getClass().getSimpleName(), hashCode(), _connector);
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests an {@link InetAddressPattern}
+     * against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     * {@link Request#getConnectionMetaData()}.
+     */
+    public static class InetAddressPatternPredicate implements Predicate<Request>
+    {
+        public static InetAddress getInetAddress(SocketAddress socketAddress)
+        {
+            if (socketAddress instanceof InetSocketAddress inetSocketAddress)
+            {
+                if (inetSocketAddress.isUnresolved())
+                {
+                    try
+                    {
+                        return InetAddress.getByName(inetSocketAddress.getHostString());
+                    }
+                    catch (UnknownHostException e)
+                    {
+                        if (LOG.isTraceEnabled())
+                            LOG.trace("ignored", e);
+                        return null;
+                    }
+                }
+
+                return inetSocketAddress.getAddress();
+            }
+            return null;
+        }
+
+        private final InetAddressPattern _pattern;
+
+        public InetAddressPatternPredicate(InetAddressPattern pattern)
+        {
+            _pattern = pattern;
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _pattern.test(getInetAddress(request.getConnectionMetaData().getRemoteSocketAddress()));
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return _pattern.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            return obj instanceof InetAddressPatternPredicate inetAddressPatternPredicate && _pattern.equals(inetAddressPatternPredicate._pattern);
+        }
+
+        @Override
+        public String toString()
+        {
+            return super.toString();
+        }
+    }
+
+    /**
+     * A {@link Predicate} over {@link Request} that tests {@link Request#getMethod() method} name.
+     * Using predicates in less efficient than using {@link ConditionalHandler#includeMethod(String...)}
+     * and {@link ConditionalHandler#excludeMethod(String...)}, so this predicate should only be used
+     * if necessary to combine with other predicates.
+     */
+    public static class MethodPredicate implements Predicate<Request>
+    {
+        private final String _method;
+
+        public MethodPredicate(String method)
+        {
+            _method = Objects.requireNonNull(method);
+        }
+
+        @Override
+        public boolean test(Request request)
+        {
+            return _method.equals(request.getMethod());
```

Comment:
Should not be `equalsIgnoreCase()`?

Reply:
No.   [HTTP methods are case-sensitive](https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.1).

Looks like our implementation of `HttpMethod#is` is insensitive!  So that is against the RFC!   Not for this PR to fix.